src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java

```diff
@@ -222,6 +223,19 @@ public FSItem retrieveById(FSContainerInterface container,
         return retrieveWorker(firestore, datasetId, enumerateDepth, fireStoreDirectoryEntry, throwOnNotFound, fileId);
     }
 
+
+    public FSItem retrieveBySnapshotandId(SnapshotProjectModel snapshot,
+                                          String fileId,
+                                          int enumerateDepth,
+                                          boolean throwOnNotFound) throws InterruptedException {
+        String projectId = snapshot.getDataProject();
+        String datasetId = snapshot.getId();
+        Firestore firestore = FireStoreProject.get(projectId).getFirestore();
+
+        FireStoreDirectoryEntry fireStoreDirectoryEntry = directoryDao.retrieveById(firestore, datasetId, fileId);
+        return retrieveWorker(firestore, datasetId, enumerateDepth, fireStoreDirectoryEntry, throwOnNotFound, fileId);
```

Comment:
nit: I know we have this pattern of using `throwOnNotFound` but I'd love for us to move to a pattern of returning optionals and handling not found logic that way.

Reply:
I agree that plumbing in a boolean to control throw behavior is upside down. It would be better to return the optional and let the caller throw or not as they please.

I looked at the blame thinking, "surely I wouldn't code it that way," but git doesn't lie. All my fault. What was I thinking?!

Reply:
lol

Reply:
should I adjust this (and all similar methods) in a follow on pr?